### ISSUE TITLE
[function-paren-newline] false positives with parentheses in type parameters

### ISSUE THREAD

**Issue by fasttime** [typescript-eslint/typescript-eslint#942]
**Repro**

```JSON
{
  "rules": {
    "function-paren-newline": ["error", "consistent"]
  }
}
```

```ts
Array<() => void>
(
);
```

or simply

```ts
Array<(any)>
(
);
```

**Expected Result**

No error.

**Actual Result**

> 3:1  error  Unexpected newline before ')'

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.1.0` |
| `@typescript-eslint/parser`        | `2.1.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.3.0` |
| `node`                             | `12.9.1` |
| `npm`                              | `6.10.2` |

**Comment by bradzacher** [comment]
Similar to #707 / #364 - this rule has no knowledge of generics.
